contracts/state-registry/StateRegistry.sol

```diff
@@ -0,0 +1,280 @@
+        varBytes32 = newVar;
+    }
+
+    function getVarBytes32() external view returns (bytes32) {
+        return varBytes32;
+    }
+
+    function setVarString(string memory newVar) external {
+        varString = newVar;
+    }
+
+    function getVarString() external view returns (string memory) {
+        return varString;
+    }
+
+    function setVarEnum(Choices newVar) external {
+        varEnum = newVar;
+    }
+
+    function getVarEnum() external view returns (Choices) {
+        return varEnum;
+    }
+
+    function setVarIntArrDataAlloc(uint[] calldata newVar) external {
+        varIntArrDataAllocBefore = newVar;
+
+        uint[] storage localVar = varIntArrDataAllocBefore;
+        localVar.pop();
+        // pointer to varIntArrDataAllocBefore, so varIntArrDataAllocAfter should pop varIntArrDataAllocBefore as well
+        varIntArrDataAllocAfter = localVar;
+    }
+
+    function getVarIntArrDataAlloc() external view returns (uint[] memory, uint[] memory) {
+        return (varIntArrDataAllocBefore, varIntArrDataAllocAfter);
+    }
+
+    function setVarIntArrDataAllocDeleted(uint[] calldata newVar) external {
+        varIntArrDataAllocDeleted = newVar;
+    }
+    function deleteVarIntArrDataAllocDeleted() external {
+        delete varIntArrDataAllocDeleted;
+    }
+
+    function getVarIntArrDataAllocDeleted() external view returns (uint[] memory) {
+        return varIntArrDataAllocDeleted;
+    }
+
+    function setVarStringConcat(string memory newVar) external {
+        varStringConcat = string.concat(varStringConcat, newVar);
+    }
+
+    function getVarStringConcat() external view returns (string memory) {
+        return varStringConcat;
+    }
+
+    function setVarStringConcatDeleted(string memory newVar) external {
+        varStringConcatDeleted = string.concat(varStringConcatDeleted, newVar);
+    }
+
+    function getVarStringConcatDeleted() external view returns (string memory) {
+        return varStringConcatDeleted;
+    }
+
+    function deleteVarStringConcatDeleted() external {
+        delete varStringConcatDeleted;
+    }
+
+    function setVarContractStruct(ContractStruct memory newVar) external {
+        varContractStruct = newVar;
+    }
+
+    function getVarContractStruct() external view returns (ContractStruct memory) {
+        return varContractStruct;
+    }
+
+     function setVarContractStructDeleted(ContractStruct memory newVar) external {
+        varContractStructDeleted = newVar;
+    }
+
+    function getVarContractStructDeleted() external view returns (ContractStruct memory) {
+        return varContractStructDeleted;
+    }
+
+    function deleteVarContractStructDeleted() external {
+        delete varContractStructDeleted;
+    }
+
+    function setBalance(address addr, uint256 value) external {
+        balance[addr] = value;
+    }
+
+    function balanceOf(address account) public view returns (uint256) {
+        return balance[account];
+    }
+}
```

Comment:
nit: one option we considered was a get function that returns all state variable values.
This would allow you to make one get call per contract before and after migration making it simpler for the state validation logic.

Reply:
another nit: we could add a function that hashes entire contract storage and we just compare the hashes before/after migration, the downside of this approach is that we can't get the differences in an easy manner
```solidity
    function getStateHash() external view returns (bytes32) {
        return
            keccak256(
                abi.encode(
                    abi.encode(
                        varUint8, varUint16, varUint32, varUint64, varUint128, varUint256, 
                        varInt8, varInt16, varInt32, varInt64, varInt128, varInt256
                    ),
                    abi.encode(
                        varBool, varAddress, varContractType, varBytes32, varString, 
                        varIntArrDataAllocBefore, varIntArrDataAllocAfter, varIntArrDataAllocDeleted
                    ),
                    abi.encode(
                        varStringConcat, varStringConcatDeleted,
                        varContractStruct, varContractStructDeleted
                    )
                )
            );
    }
 ```